### PR TITLE
[Test][Enhancement] Modularize signup function for further testing

### LINKED ISSUE #41: Modularize Login/SignUp Test Function
Modularize signUp functions and modify the test to protect the signUp logic 
- Modularize signUp functions
- Call signUp function in test methods
- pass the test

### PR DESCRIPTION
**Pull request by Lisooo790926**
closed #41 

## Summary

Modularizing user related function for further testing
- loginOrInitUser
- signUp

## Impacted Areas

All related testing

## Tests
Post test pass

## Visual Materials
image

### CODE REVIEW

**Review by NOOMA-42 — APPROVED** [review]
TODO: // TODO remove this method, once successfully mock the TwitterClient result
    async loginOrInitUserForTest(userId: string): Promise<User>

**Comment by ryanycw** [comment]
Thank you for the request! I will review it by the end of the day.

File: packages/relay/src/services/UserService.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { UnirepSocialSynchronizer } from '../synchornizer'
+import TransactionManager from '../singletons/TransactionManager'
+import { APP_ADDRESS, CLIENT_URL } from '../config'
+import TwitterClient from '../singletons/TwitterClient'
+import crypto from 'crypto'
+import { UserRegisterStatus } from '../enums/userRegisterStatus'
+import { Contract } from 'ethers'
+import User from '../data/User'
+
+const STATE = 'state'
```
Comment (ryanycw) [inline packages/relay/src/services/UserService.ts:12]: Minor issue, why do we declare STATE here. Can we check state == 'state' direcly?

File: packages/relay/src/services/UserService.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            throw Error("wrong callback value")
+
+        var userInfo = await TwitterClient.authClient.requestAccessToken(code as string)
+            .then((_) => TwitterClient.client.users.findMyUser())
+
+        const userId = userInfo.data?.id!!
+        return this.loginOrInitUserForTest(userId)
+    }
+
+    // TODO remove this method, once successfully mock the TwitterClient result
```
Comment (ryanycw) [inline packages/relay/src/services/UserService.ts:37]: Love this! @cc03668 We can put this into back log.

File: packages/relay/src/services/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        const appContract = TransactionManager.appContract!!
+        const calldata = appContract.interface.encodeFunctionData('userSignUp', [
+            signupProof.publicSignals,
+            signupProof.proof,
+            hashUserId,
+            fromServer,
+        ])
+
+        // TODO: fix transction twice bug
```
Comment (ryanycw) [inline packages/relay/src/services/UserService.ts:97]: No, need to fix it here. Just noting that this could be cleaned up.

**Review by ryanycw — COMMENTED** [review]
LGTM! Only some minor issues we should confirm

**Comment by ryanycw** [comment]
I will merge this so that following tests can proceed. We can resolve the above issues after the merge.

**Pull request merged**